src/Microsoft.AspNet.Http.Extensions/SendFileResponseExtensions.cs

```diff
@@ -74,10 +74,56 @@ public static Task SendFileAsync(this HttpResponse response, string fileName, lo
             var sendFile = response.HttpContext.Features.Get<IHttpSendFileFeature>();
             if (sendFile == null)
             {
-                throw new NotSupportedException(Resources.Exception_SendFileNotSupported);
+                return SendFileAsync(response.Body, fileName, offset, count, cancellationToken);
             }
 
             return sendFile.SendFileAsync(fileName, offset, count, cancellationToken);
         }
+
+        // Not safe for overlapped writes.
+        private static async Task SendFileAsync(Stream outputStream, string fileName, long offset, long? length, CancellationToken cancel)
+        {
+            cancel.ThrowIfCancellationRequested();
+
+            if (string.IsNullOrWhiteSpace(fileName))
+            {
+                throw new ArgumentNullException(nameof(fileName));
+            }
+            if (!File.Exists(fileName))
+            {
+                throw new FileNotFoundException(string.Empty, fileName);
+            }
+
+            var fileInfo = new FileInfo(fileName);
+            if (offset < 0 || offset > fileInfo.Length)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset), offset, string.Empty);
+            }
+
+            if (length.HasValue &&
+                (length.Value < 0 || length.Value > fileInfo.Length - offset))
+            {
+                throw new ArgumentOutOfRangeException(nameof(length), length, string.Empty);
+            }
+
+            var fileStream = new FileStream(
+                fileName,
+                FileMode.Open,
+                FileAccess.Read,
+                FileShare.ReadWrite,
+                bufferSize: 1024 * 64,
```

Comment:
This is a 64kb buffer, but StreamCopyOperation uses a 16kb buffer. Would it help if they were aligned? Do we know if this is a managed or native buffer?